sigma-test.go

```diff
@@ -147,7 +148,21 @@ func testFile(path string, r sigma.Rule, configs []sigma.Config) (error, []strin
 		return errNoTests, nil
 	}
 
-	rule := evaluator.ForRule(r, evaluator.WithConfig(configs...), evaluator.WithPlaceholderExpander(func(ctx context.Context, placeholderName string) ([]string, error) {
+	// only use logsources that are relevant for this rule. This avoids having conflicts with other logsources with the same field names
+	var relevantConfigs []sigma.Config
+	for _, c := range configs {
+		for _, v := range c.Logsources {
+			if (v.Logsource.Product == r.Logsource.Product || v.Rewrite.Product == r.Logsource.Product) && (v.Logsource.Category == r.Logsource.Category || v.Rewrite.Category == r.Logsource.Category) {
```

Comment:
Rewriting is complicated so let's just ignore it for now? (it actually needs to mutate the product/category/etc. https://github.com/bradleyjkemp/sigma-go/blob/99be942d010e9b5f99247761da320974e3efe304/evaluator/indexes.go#L35-L45)

I also think the logic is too strict here? Rules don't _have_ to have both a product and category. As long as the logsource doesn't conflict with the rule, it's relevant? (at least that's sigma-go's interpretation https://github.com/bradleyjkemp/sigma-go/blob/99be942d010e9b5f99247761da320974e3efe304/evaluator/indexes.go#L23-L32)